src/components/widgets/AutocompleteWidget/AutocompleteWidget.tsx

```diff
@@ -121,83 +125,126 @@ function AutocompleteWidget(props: AutocompleteWidgetProps) {
     };
 
     /**
-     * If a selectOption is provided, we obtain the label via API
+     * on mount: fetches term for selectOption and sets it's label or sets a given label if no iri is provided or the given iri cannot be resolved only if allowCustomTerms is true
      */
-    useEffect(() => {
-        if (props.selectOption?.iri && props.selectOption?.iri.startsWith("http")) {
-            setLoadingState(true);
-            olsApi.select(
-                {query: props.selectOption?.iri},
-                undefined,
-                undefined,
-                props.parameter,
-            ).then((response) => {
-                if (response.response && response.response.docs) {
-                    response.response.docs.map((selection: any) => {
-                        if (props.selectOption?.iri === selection.iri) {
-                            setOptions([
-                                {
-                                     // label to display within the combobox either raw value or generated one
-                                    // #renderOption() is used to display during selection.
-                                    label: hasShortSelectedLabel ? selection.label : generateDisplayLabel(selection),
-                                    value: {
-                                        iri: selection.iri,
-                                        label: selection.label,
-                                        ontology_name: selection.ontology_name,
-                                        type: selection.type,
-                                        short_form: selection.short_form,
+    const {
+        isLoading: isLoadingOnMount
+    } = useQuery(
+        [
+            "onMount" // no dependencies - does only need to be executed once when mounting the component @TODO: Dependency on props.selectOption
```

Comment:
Should this TODO be addressed before merging?

Reply:
It already existed before, so I kept it because I did not really get the meaning of it. Is there a difference in including selectOption in the queryKey? I thought not because selectOption is a preselected term, and changes to it will only show in the widget when it is remounted, but in this case the useQuery hook will be executed anyway.